// scldb/index.js
var Database = require('better-sqlite3');
var db = new Database(__dirname+'/db/scl.db');

// === soc ===

exports.idSoc=function(id){
var stm=db.prepare('select * from soc where id=?');
var row=stm.get(id);
return row;
};

exports.namSoc=function(arg2){
var stm=db.prepare('select * from soc where nam=?');
try{var row=stm.get(arg2);
}catch(err){console.log(err);}
return row;
};

exports.allSoc=function(){
var stm=db.prepare('select * from soc ');
var row=stm.all();
return row;
};

// === soc run ===

exports.insSoc=function(id,prv,nam,dis){
var stm=db.prepare('insert into soc values(?,?,?,?) ');
stm.run(id,prv,nam,dis)
};

exports.updSoc=function(uni,soc,sku){
var stm=db.prepare('update soc set uni=? where soc=? and sku=? ');
try{stm.run(uni,soc,sku);
}catch(err){console.log(err);}
};

exports.delSoc=function(){
var stm=db.prepare('delete from soc where uni=0');
try{stm.run();
}catch(err){var err=err;console.log(err);}
};

// === det ===

exports.idDet=function(id){
var stm=db.prepare('select * from det where id=?');
var row=stm.get(id);
return row;
};

exports.namDet=function(arg2){
var stm=db.prepare('select * from det where nam=?');
try{var row=stm.all(arg2);
}catch(err){console.log(err);}
return row;
};

exports.allImg=function(){
var stm=db.prepare('select img from det ');
var row=stm.all();
return row;
};
// === det run ===

exports.insDet=function(id,img,ema,des){
var stm=db.prepare('insert into det values(?,?,?,?) ');
stm.run(id,img,ema,des)
};

exports.updDet=function(uni,det,sku){
var stm=db.prepare('update det set uni=? where det=? and sku=? ');
try{stm.run(uni,det,sku);
}catch(err){console.log(err);}
};

exports.delDet=function(){
var stm=db.prepare('delete from det where uni=0');
try{stm.run();
}catch(err){var err=err;console.log(err);}
};

// === pid ===

exports.insPid=function(nam,pid,mnt,buy,ite,tim){
var stm=db.prepare('insert into pid values(?,?,?,?,?,?)');
stm.run(nam,pid,mnt,buy,ite,tim)
};

exports.getPid=function(nam){
var stm=db.prepare('select *,rowid from pid where nam=? order by rowid desc ');
try{var row=stm.get(nam);
}catch(err){console.log(err);}
return row;
};

exports.allPid=function(nam){
var stm=db.prepare('select *,rowid from pid where nam=? order by rowid desc ');
try{var row=stm.all(nam);
}catch(err){console.log(err);}
return row;
};

exports.selPid=function(pid){
var stm=db.prepare('select *,rowid from pid where nam=?');
try{var row=stm.get(pid);
}catch(err){console.log(err);}
return row;
};


exports.maxPid=function(){
var stm=db.prepare('select rowid from pid order by rowid desc');
try{var row=stm.get();
}catch(err){console.log(err);}
return row;
};

exports.idPid=function(id){
var stm=db.prepare('select * from pid where rowid=?');
try{var row=stm.get(id);
}catch(err){console.log(err);}
return row;
};


// === pal run ===

exports.insPal=function(nam,id,mnt,ite,utc){
var stm=db.prepare('insert into pal values(?,?,?,?,?)');
stm.run(nam,id,mnt,ite,utc)
};

exports.itePal=function(ite,tok){
var stm=db.prepare('update pal set ite=? where tok=?');
stm.run(ite,tok);
};

// === pal get ===
exports.getPal=function(nam){
var stm=db.prepare('select *,rowid from pal where nam=? order by rowid desc ');
var row=stm.get(nam);
return row;
};

exports.allPal=function(nam){
var stm=db.prepare('select *,rowid from pal where nam=? order by rowid desc ');
var row=stm.all(nam);
return row;
};
