// cardb/index.js
var Database = require('better-sqlite3');
var db = new Database(__dirname+'/db/car.db');

// === mer ===

exports.catMer=function(cat){
var stm=db.prepare('select * from mer where cat=?');
try{var row=stm.get(cat);}
catch(err){console.log(err);}
return row;
};

exports.allMer=function(){
var stm=db.prepare('select *,rowid from mer order by sku desc');
try{var row=stm.all();
}catch(err){console.log(err);}
return row;
};

exports.skuMer=function(sku){
var stm=db.prepare('select * from mer where sku=?');
try{var row=stm.get(sku);}
catch(err){console.log(err);}
return row;
};

exports.insMer=function(name,sku,pri,des){
var stm=db.prepare('insert into mer values(?,?,?,?)');
stm.run(name,sku,pri,des);
};

exports.delMer=function(sku){
var stm=db.prepare('delete from mer where sku=?')
stm.run(sku)
};
exports.namMer=function(name,sku){
var stm=db.prepare('update mer set name=? where sku=?')
try{stm.run(name,sku)}
catch(err){console.log(err)}
};
exports.priMer=function(pri,sku){
var stm=db.prepare('update mer set pri=? where sku=?')
stm.run(pri,sku)
};

exports.desMer=function(des,sku){
var stm=db.prepare('update mer set des=? where sku=?')
stm.run(des,sku)
};

// === tmp ===
exports.mailTmp=function(arg2){
var stm=db.prepare('select * from tmp where email=?');
try{var row=stm.all(arg2);
}catch(err){console.log(err);}
return row;
};

exports.allTmp=function(){
var stm=db.prepare('select * from tmp ');
var row=stm.all();
return row;
};

exports.skuTmp=function(sku){
var stm=db.prepare('select * from tmp where sku=?');
try{var row=stm.get(sku);
}catch(err){console.log(err);}
return row;
};

exports.usrTmp=function(usr){
var stm=db.prepare('select * from tmp where usr=?');
var row=stm.get(usr);
return row;
};

// === tmp run ===
exports.updTmp=function(uni,email,sku){
var stm=db.prepare('update tmp set uni=? where email=? and sku=? ');
try{stm.run(uni,email,sku);
}catch(err){console.log(err);}
};

exports.insTmp=function(email,sku,uni){
var stm=db.prepare('insert into tmp values(?,?,?) ');
try{stm.run(email,sku,uni);
}catch(err){console.log(err);}
};

exports.delUni=function(){
var stm=db.prepare('delete from tmp where uni=0');
try{stm.run();
}catch(err){console.log(err);}
};

exports.delEma=function(email){
var stm=db.prepare('delete from tmp where email=?');
try{stm.run(email);
}catch(err){console.log(err);}

};
